File: FileSharing/FileSharing/Models/FileInfoModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FileSharing.DAL.Entity
{
    public class FileInfoModel
    {
        public int Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Path { get; set; } = string.Empty;
        public string Extension { set; get; } = string.Empty;
        public long Size { get; set; }
        public string ContentType { get; set; } = string.Empty;
        public bool CanBeDownloaded { get; set; }
        public int DownloadCount { get; set; } = 0;
        public DateTime UploadDate { get; set; }
        public DateTime LastDownloadDate { get; set; }
        public string CategoryTitle { get; set; }

        public int Category_Id { get; set; }
    }
}

File: FileSharing/FileSharing/ClientApp/src/app/Entity/FileInfoModel.ts
export class FileInfoModel {
  constructor(Id: number, Title: string, Path: string, Extension: string, Size: number, ContentType: string,
    CanBeDownloaded: boolean, category_Id: number, categoryTitle: string, lastDownloadDate: Date, uploadDate: Date)
  {
    this.id = Id;
    this.title = Title;
    this.path = Path;
    this.size = Size;
    this.extension = Extension;
    this.contentType = ContentType;
    this.canBeDownloaded = CanBeDownloaded;
    this.categoryTitle = categoryTitle;
    this.category_Id = category_Id;
    this.uploadDate = uploadDate;
    this.lastDownloadDate = lastDownloadDate;
  }

  public id: number;
  public title: string;
  public path: string;
  public extension: string;
  public size: number;
  public contentType: string;
  public canBeDownloaded: boolean;
  public uploadDate: Date;
  public lastDownloadDate: Date;
  public downloadCount: number = 0;
  public categoryTitle: string;
  public category_Id: number;
}
